.gitignore

```diff
@@ -1,29 +1,33 @@
+# Compile products
 *.pyc
 *.class
-*.log
-dl/src/main/python/.idea/
+target/
+dist/
```

Comment:
Ack @yiheng I made a mistake here. This still actually matches the `spark/dist` directory. I should have changed it to `/dist/` to only match the root dist directory. I'm sorry about that, I'll open another PR to correct it.